src/mds/CDir.cc

```diff
@@ -2922,8 +2935,6 @@ void CDir::scrub_initialize()
     }
   }
   scrub_infop->directory_scrubbing = true;
-
-  assert(scrub_local()); // TODO: handle failure
```

Comment:
On Wed, Feb 17, 2016 at 2:53 AM, Gregory Farnum <notifications@github.com>
wrote:

> In src/mds/CDir.cc
> <https://github.com/ceph/ceph/pull/6682#discussion_r53058563>:
>
> > @@ -2922,8 +2935,6 @@ void CDir::scrub_initialize()
> >      }
> >    }
> >    scrub_infop->directory_scrubbing = true;
> > -
> > -  assert(scrub_local()); // TODO: handle failure
>
> Well, it's *very important* we have all the directory's contents in
> memory when we set up the scrub metadata, because otherwise we'll be
> skipping children. Obviously that should happen, but scrub_local() was a
> check to guarantee it. Given the low odds of otherwise detecting missing
> scrubs of descendants, that's a check I think we need to keep around unless
> there's a compelling reason not to.
>
There already is is_complete() check in CDir::scrub_initialize().  Inode's
rstat and fragstat can be wrong, so we can use assert() here. without
assert() I can't see how scrub_local() makes extra guarantee

—

> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/ceph/pull/6682/files#r53058563>.
>